trio/_tests/test_highlevel_ssl_helpers.py

```diff
@@ -33,24 +41,48 @@ async def echo_handler(stream):
 # you ask for.
 @attr.s
 class FakeHostnameResolver(trio.abc.HostnameResolver):
-    sockaddr = attr.ib()
-
-    async def getaddrinfo(self, *args):
+    sockaddr: tuple[str, int] | tuple[str, int, int, int] = attr.ib()
+
+    async def getaddrinfo(
+        self,
+        host: bytes | str | None,
+        port: bytes | str | int | None,
+        family: int = 0,
+        type: int = 0,
+        proto: int = 0,
+        flags: int = 0,
+    ) -> list[
+        tuple[
+            AddressFamily,
+            SocketKind,
+            int,
+            str,
+            tuple[str, int] | tuple[str, int, int, int],
+        ]
+    ]:
         return [(AF_INET, SOCK_STREAM, IPPROTO_TCP, "", self.sockaddr)]
 
-    async def getnameinfo(self, *args):  # pragma: no cover
+    async def getnameinfo(self, *args: Any) -> NoReturn:  # pragma: no cover
```

Comment:
`object`?

Reply:
it might raise a type error about the signature not matching the superclass, not sure. But with an empty body it doesn't matter either way.